### PR TITLE
Update pull_request_template

### PR DESCRIPTION
**Pull request by coliff**
- Remove mention about needing to update CHANGELOG (no longer needed)
- Comment out instructions to prevent them appearing in every PR which adds noise

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (including CHANGELOG updates)
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,8 +1,6 @@
 ## Description
 
-Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
-
-Fixes # (issue)
```
Comment (ryan-henness-trimble) [inline .github/pull_request_template.md:5]: Should we keep this? This automatically links the PR to the corresponding work item - maybe we should go with `References #`

Comment (coliff) [reply .github/pull_request_template.md:5]: yep, good idea. :+1 The reason I removed it is for all PRs I make I create branch from the GitHub Web UI (from the Development section on the right) and it auto-links the PR already without needing to type in the related issue number... but I understand that not everyone will do that.

Comment (msankaran0712) [reply .github/pull_request_template.md:5]: To me "Fixes" sounds more appropriate, but not a big issue.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -13,14 +13,14 @@ Fixes # (issue)
 
 ## How Has This Been Tested?
 
-Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
+<!--Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->
```
Comment (msankaran0712) [inline .github/pull_request_template.md:16]: I think "Provide instructions so we can reproduce. Please also list any relevant details for your test configuration" is still relevant, not sure why it was removed.

**Review by msankaran0712 — APPROVED** [review]

**Pull request merged**